docs/fundamentals/executor/monitoring-executor.md

```diff
@@ -0,0 +1,151 @@
+(monitoring-executor)=
+# Monitor Executor with Custom Metrics
```

Comment:
Maybe we should explain how this feature works in the abstract (only a few lines) before going into the example?